### PR TITLE
Feature/apply if and apply if async operators

### PR DESCRIPTION
**Pull request by jakub-hajduk**

### CODE REVIEW

**Comment by jakub-hajduk** [comment]
there is also another one - toggle status.

File: apply-if-async/apply-if-async-operator.spec.ts [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+      });
+  });
+
+  it('should map value to \'falsy\' if given observable returned false', () => {
+    getObs$(false).subscribe((value: string) => {
+        expect(value).toEqual('falsy');
+      });
+  });
+
+  // it('should apply given elseOperators if given observable returned false.', () => {
```
Comment (dbojarski) [inline apply-if-async/apply-if-async-operator.spec.ts]: remove please

File: apply-if-async/apply-if-async.operator.ts [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+
+import { Operators } from '../shared/types/operators';
+import { castArray } from '../shared/utils/cast-array';
+
+/**
+ * Applies truthyOperators to the pipe if observable returns true. Otherwise applies falsyOperators
+ */
+export function applyIfAsync<T>(observable: Observable<boolean>, truthyOperators: Operators = [], falsyOperators: Operators = []): MonoTypeOperatorFunction<T> {
+    return (source: Observable<T>): Observable<T> => observable.pipe(
+        // @ts-ignore
```
Comment (dbojarski) [inline apply-if-async/apply-if-async.operator.ts]: remove or add tslint fle

File: toggle-status/toggle-status-operator.spec.ts [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+
+describe('toggle-status operator', () => {
+  it('should toggle status values to busy=true and done=false while observable is pending', (done) => {
+
+    const status: TogglableStatus = <TogglableStatus>{};
+
+    const request = of('value').pipe(delay(200));
+
+    request
+    .pipe(
```
Comment (dbojarski) [inline toggle-status/toggle-status-operator.spec.ts]: format

**Review by dbojarski — CHANGES_REQUESTED** [review]

**Pull request merged**